### PR TITLE
Feature/toggle-campus-button

### PR DESCRIPTION
**Pull request by muhanator**
This PR references #12 where we now add the ability to toggle between the two campuses along with a refactored map that is now going to be changed upon using changenotifier, changenotifierprovider and consumer

### CODE REVIEW

**Comment by ooms97** [comment]
We need some unit tests for the lib/services/map_helper.dart file.

File: test/map_helper_test.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import 'package:concordi_around/services/map_helper.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:concordi_around/services/constants.dart' as constants;
```
Comment (ooms97) [inline test/map_helper_test.dart]: Remove this unused import.

File: lib/main.dart [hunk 2]
```diff
@@ -10,7 +10,12 @@ class MyApp extends StatelessWidget {
...
         title: 'ConcordiAround',
-        home: ChangeNotifierProvider(
-            child: HomePage(), create: (context) => MapNotifier()));
+        home: MultiProvider(
+          providers: [
+            ChangeNotifierProvider(create: (context) => MapNotifier() )
+          ],
+          child: HomePage()
+        )
+    );
```
[the remark this answers is not in the document]
Comment (muhanator) [reply lib/main.dart]: Yes just fixed

File: lib/views/home_page.dart [hunk 3]
```diff
@@ -14,8 +14,6 @@ class _HomePageState extends State<HomePage> {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-        body: Map(), 
-        drawer: SidebarDrawer(), 
-        resizeToAvoidBottomInset: false);
+        body: Map(), drawer: SidebarDrawer(), resizeToAvoidBottomInset: false);
```
[the remark this answers is not in the document]
Comment (muhanator) [reply lib/views/home_page.dart]: done!

File: lib/services/map_helper.dart [hunk 4]
```diff
@@ -14,7 +14,26 @@ class MapHelper {
...
 
+  static bool isWithinSGW(LatLng latLng) {
+    List<LatLng> coordsList = [
+      LatLng(45.492532, -73.576848),
+      LatLng(45.496587, -73.581582),
+      LatLng(45.498974, -73.579218),
+      LatLng(45.496573, -73.572700),
+    ];
+    return boundsFromLatLngList(coordsList).contains(latLng);
+  }
```
[the remark this answers is not in the document]
Comment (muhanator) [reply lib/services/map_helper.dart]: Yes i just added them with 100% line coverage

File: lib/services/map_helper.dart [hunk 5]
```diff
@@ -14,7 +14,26 @@ class MapHelper {
...
+
+  static bool isWithinLoyola(LatLng latLng) {
+    List<LatLng> coordsList = [
+      LatLng(45.458355, -73.633476),
+      LatLng(45.459243, -73.640739),
+      LatLng(45.457324, -73.642574),
+      LatLng(45.455706, -73.638207)
+    ];
```
[the remark this answers is not in the document]
Comment (muhanator) [reply lib/services/map_helper.dart]: done!

File: test/map_helper_test.dart (hunk 1 above)
Comment (muhanator) [reply test/map_helper_test.dart]: fixed!

File: lib/provider/map_notifier.dart [hunk 6]
```diff
@@ -49,4 +65,15 @@ class MapNotifier with ChangeNotifier {
...
   }
-}
+
+  Future<void> toggleCampus(LatLng latLng) async {
+    final c = await _completer.future;
+    final p = CameraPosition(
+      target: latLng,
+      zoom: constants.CAMERA_DEFAULT_ZOOM,
+      tilt: constants.CAMERA_DEFAULT_TILT,
+      bearing: 30.8334901395799,
```
Comment (ooms97) [inline lib/provider/map_notifier.dart]: Why change the bearing from default on the toggle button? Seems like a strange and unwanted behavior.

File: lib/provider/map_notifier.dart [hunk 7]
```diff
@@ -29,6 +31,20 @@ class MapNotifier with ChangeNotifier {
     notifyListeners();
   }
 
+  void setCampusLatLng(LatLng latLng) {
```
Comment (ChesterLY) [inline lib/provider/map_notifier.dart]: - As long as the user isn't within Loyola, they are within SGW. Is that the desired result?

File: lib/services/map_helper.dart [hunk 8]
```diff
@@ -14,8 +16,33 @@ class MapHelper {
       LatLng(45.49741, -73.57819),
       LatLng(45.49781, -73.57906)
     ];
+    return (latLng == null
+        ? false
+        : boundsFromLatLngList(coordsList).contains(latLng));
+  }
 
-    return boundsFromLatLngList(coordsList).contains(latLng);
+  static bool isWithinSGW(LatLng latLng) {
```
Comment (ChesterLY) [inline lib/services/map_helper.dart]: - isWithinSGW is completely unused except for testing, should this be part of the conditions in the map_notifier.dart class?

File: lib/provider/map_notifier.dart (hunk 7 above)
Comment (muhanator) [reply lib/provider/map_notifier.dart]: Yes by default the value of the toggle remains sgw as long as the user does not enter loyola by going to it via the map or clicking the toggle to go to it

File: lib/services/map_helper.dart (hunk 8 above)
Comment (muhanator) [reply lib/services/map_helper.dart]: yes this has now been removed, good catch

File: lib/provider/map_notifier.dart (hunk 6 above)
Comment (muhanator) [reply lib/provider/map_notifier.dart]: You're right this can automatically be done by a double swipe on the map so there is no need for it here.
Removed!

**Pull request merged**